include/seqan3/alphabet/container/concatenated_sequences.hpp

```diff
@@ -1351,6 +1251,88 @@ class concatenated_sequences
 
+    /*!\brief Appends an empty element to the end of the container.
+     * \tparam rng_type The type of range to be inserted; must satisfy \ref is_compatible_with_value_type.
+     * \param value The value to append.
+     *
+     * This conceptionally adds an element to the container,
+     * i.e. the "outer container" grows by 1 and the new back() will be empty.
+     *
+     * No iterators are invalidated.
+     *
+     * ### Complexity
+     *
+     * Amortised constant. Wort-case linear in size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container my contain invalid data after exceptions is
+     * thrown.
+     *
+     * \experimentalapi{Experimental since version 3.1.}
+     */
+    void push_back()
+    {
+        data_delimiters.push_back(data_delimiters.back());
+    }
+
+    /*!\brief Appends the given element-of-element value to the end of the container.
+     * \param value The value to append.
+     *
+     * This conceptionally performs a `push_back()` on the `back()` of this container,
+     * i.e. that last inner container grows by 1.
+     *
+     * If the new concat_size() is greater than concat_capacity() then all iterators and references (including the past-the-end
+     * iterator) are invalidated. Otherwise only the past-the-end iterator is invalidated.
+     *
+     * ### Complexity
+     *
+     * Amortised constant. Wort-case linear in concat_size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container my contain invalid data after exceptions is
+     * thrown.
+     *
+     * \experimentalapi{Experimental since version 3.1.}
+     */
+    void push_back_back(std::ranges::range_value_t<inner_type> const value)
+    {
+        data_values.push_back(value);
+        ++data_delimiters.back();
+    }
+
+    /*!\brief Appends the given elements to the end of the container (increases size of last element by n).
+     * \tparam rng_type The type of range to be inserted; must satisfy \ref is_compatible_with_value_type.
+     * \param value The value to append.
+     *
+     * This conceptionally performs an `insert()` on the `back()` of this container,
+     * i.e. the last inner container grows by value.size().
+     *
+     * If the new concat_size() is greater than concat_capacity() then all iterators and references (including the past-the-end
+     * iterator) are invalidated. Otherwise only the past-the-end iterator is invalidated.
+     *
+     * ### Complexity
+     *
+     * Amortised linear in the size of value. Wort-case linear in concat_size().
+     *
+     * ### Exceptions
+     *
+     * Basic exception guarantee, i.e. guaranteed not to leak, but container my contain invalid data after exceptions is
+     * thrown.
+     *
+     * \experimentalapi{Experimental since version 3.1.}
+     */
+    template <std::ranges::forward_range rng_type>
+    void append_back(rng_type && value)
```

Comment:
Since there is an _inner_ `push_back_back(...)` that corresponds to the _outer_ `push_back(...)`, maybe it would be nice to have and _outer_ version of this function, e.g. `append(...)`. This could prevent confusion, because `append_back(...)` sounds similar to `push_back(...)`, but is actually an _inner_ fucntion like `push_back_back()`.

This technically exists already with some more flexibilty in the form of  `insert(...)`. So maybe it would be nice to have a directly corrensponding set of `insert_back()` overloads. However, this would likely not be ergonomic, due to the iterator-based API of `insert(...)` and the fact that we return a view as _outer_ elements.

So I guess we stick with this ^^

Reply:
This is exactly my problem at the moment. The proposed changes do not fit well into the current design of this data structure. Irrespective of the naming. Nor do they fit into any known interface of similar structures from the STL. I am not saying that this is necessarily a bad thing. But if the proposed changes are important in order to use it efficiently then the semantic meaning of this structure should be reconsidered. Or maybe a different structure is needed.